### ISSUE TITLE
Grouped Folders

### ISSUE THREAD

**Issue by marbetschar** [elementary/mail#449]
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Feature

If we do have multiple email accounts in elementary Mail it would be convenient to add grouped folders - at least for well known folder names. That is: Inbox, Archive, Drafts, Sent. I would expect these folders to be shown at the very top, containing the messages from all configured email accounts. This way we can get one single Inbox folder, showing an overview for all received messages, one single Archive folder showing all archived messages, ... etc.

**Existing work**
Basically everything is there. Would just need an additional check if there are multiple accounts, and if so show another, predefined group of folders which are linked to the corresponding folders in all accounts.
